tests/bluetooth/tester/src/audio/btp_pbp.c

```diff
@@ -0,0 +1,288 @@
+/* btp_pbp.c - Bluetooth PBP Tester */
+
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "btp/btp.h"
+
+#include <zephyr/bluetooth/audio/pbp.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#define LOG_MODULE_NAME bttester_pbp
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_BTTESTER_LOG_LEVEL);
+
+#define PBP_EXT_ADV_METADATA_LEN_MAX 128
+#define PBP_ADV_HEADER_SIZE (BT_UUID_SIZE_16 + sizeof(uint8_t) * 2)
+
+static uint8_t pbp_features_cached;
+static uint8_t pbp_metadata_cached[PBP_EXT_ADV_METADATA_LEN_MAX];
+static uint8_t pbp_metadata_cached_len;
+static uint8_t pbp_broadcast_name_cached[BT_AUDIO_BROADCAST_NAME_LEN_MAX];
+static uint8_t pbp_name_cached_len;
+
+static bool scan_get_broadcast_name_len(struct bt_data *data, void *user_data)
+{
+	uint8_t *broadcast_name_len = user_data;
+
+	switch (data->type) {
+	case BT_DATA_BROADCAST_NAME:
+		*broadcast_name_len = data->data_len;
+		return false;
+	default:
+		return true;
+	}
+}
+
+static bool scan_get_data(struct bt_data *data, void *user_data)
+{
+	enum bt_pbp_announcement_feature source_features;
+	uint32_t broadcast_id;
+	uint8_t *metadata;
+	struct bt_uuid_16 adv_uuid;
+	struct btp_pbp_ev_public_broadcast_anouncement_found_rp *ev = user_data;
+
+	switch (data->type) {
+	case BT_DATA_BROADCAST_NAME:
+		ev->broadcast_name_len = data->data_len;
+		memcpy(ev->broadcast_name, data->data, data->data_len);
+		return true;
+	case BT_DATA_SVC_DATA16:
+		if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+			return true;
+		}
+		if (!bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
+			broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+			sys_put_le24(broadcast_id, ev->broadcast_id);
+			return true;
+		}
+
+		int ret = bt_pbp_parse_announcement(data, &source_features, &metadata);
+
+		if (ret >= 0) {
+			ev->pba_features = source_features;
+			return true;
+		}
+
+		return true;
+	default:
+		return true;
+	}
+}
+
+static void pbp_scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) ||
+	    !(info->adv_props & BT_GAP_ADV_PROP_EXT_ADV) || info->interval == 0) {
+		return;
+	}
+
+	uint8_t broadcast_name_len = 0;
+	struct net_buf_simple ad_copy;
+
+	net_buf_simple_clone(ad, &ad_copy);
+	bt_data_parse(&ad_copy, scan_get_broadcast_name_len, &broadcast_name_len);
+
+	struct btp_pbp_ev_public_broadcast_anouncement_found_rp *ev_ptr;
+
+	tester_rsp_buffer_lock();
+	tester_rsp_buffer_allocate(sizeof(*ev_ptr) + broadcast_name_len, (uint8_t **)&ev_ptr);
+
+	sys_put_le24(0xffffff, ev_ptr->broadcast_id);
+	ev_ptr->pba_features = 0xff;
+	ev_ptr->broadcast_name_len = 0xff;
```

Comment:
I would suggest to do
```suggestion
	ev_ptr->pba_features = 0U;
	ev_ptr->broadcast_name_len = 0U;
```

As 0 are invalid values for both of them, and we do not use "reserved for future use" values

Reply:
Done